crates/iceberg/src/metadata_scan.rs

```diff
@@ -95,7 +99,17 @@ impl<'a> SnapshotsTable<'a> {
     }
 
     /// Scans the snapshots table.
-    pub fn scan(&self) -> Result<RecordBatch> {
+    pub async fn scan(&self) -> Result<ArrowRecordBatchStream> {
+        let arrow_schema = Arc::new(self.schema());
+        let table_metadata = self.table.metadata_ref();
+
+        Ok(
+            futures::stream::once(async move { Self::build_batch(arrow_schema, &table_metadata) })
```

Comment:
Hi, maybe we can use [`stream::iter`](https://docs.rs/futures/latest/futures/stream/fn.iter.html) instead to convert an Iterator into stream directly. The `async move {}` call here seems not needed.

Reply:
Sorry, I'm not sure how to do that with `stream::iter`.

`stream::iter` takes an iterator but not sure what we would, conceptually, we iterate over (even if just a single element). We can try create an iterator from a closure (something like [`iter::from_fn`](https://doc.rust-lang.org/1.82.0/std/iter/fn.from_fn.html)) from which we return once but that seems more verbose.

The `async move` seem necessary to me (and the compiler) because, as I understand, `once` takes an async function and `move` allows the closure to own `arrow_schema` and `table_metadata` so it can outlive the caller.

Reply:
I do agree that `futures::stream::once(async move { ... ` is a bit verbose.

I went with `try_stream!` instead, so now we do this:

```rust
pub async fn scan(&self) -> Result<ArrowRecordBatchStream> {
    let arrow_schema = Arc::new(self.schema());
    ...

    Ok(try_stream! {
        ...

        yield RecordBatch::try_new(arrow_schema, vec![...])?;
    }
    .boxed())
}
```

@xuanwo, I don't know if that's what you had in mind and if you think it's better. I'm happy to revert. Let me know.

Reply:
For consistency, I also replaced the `future::stream::once(async move { ...` with `try_stream!` in the manifests table.

Again, happy to revert.

Reply:
`stream::iter([batch])` should just work?

Reply:
Changed to `stream::iter([batch])`.